src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/Platform/AzureManagedServiceAuthenticator.cs

```diff
@@ -21,17 +22,30 @@ public AzureManagedServiceAuthenticator(bool useAzureManagedIdentity, string azu
             _useAzureManagedIdentity = useAzureManagedIdentity;
             _azureServiceTokenProviderResource = azureServiceTokenProviderResource;
             _tenantId = tenantId;
-            _azureServiceTokenProvider = new AzureServiceTokenProvider();
+            _defaultAzureCredential = new DefaultAzureCredential(new DefaultAzureCredentialOptions
+            {
+                ExcludeEnvironmentCredential = true,
+                ExcludeManagedIdentityCredential = true,
+                ExcludeVisualStudioCredential = true,
+                ExcludeAzureCliCredential = true,
+                ExcludeAzurePowerShellCredential = true,
+                ExcludeSharedTokenCacheCredential = true,
+                InteractiveBrowserTenantId = _tenantId
+            });
         }
 
-        public Task<string> GetAuthenticationToken()
+        public async Task<string> GetAuthenticationToken()
         {
             if (!_useAzureManagedIdentity)
             {
-                return Task.FromResult((string)null);
+                return await Task.FromResult((string)null).ConfigureAwait(false);
             }
 
-            return _azureServiceTokenProvider.GetAccessTokenAsync(_azureServiceTokenProviderResource, _tenantId);
+            var accessToken = await _defaultAzureCredential.GetTokenAsync(
+                new TokenRequestContext(new[] { $"{_azureServiceTokenProviderResource}.default" }) { }
+            ).ConfigureAwait(false);
+
+            return !string.IsNullOrWhiteSpace(accessToken.Token) ? accessToken.Token : null;
```

Comment:
Is this nullcheck useful? It returns null when the variable is null. If it is empty, it also returns null but in this case subsequent MS library calls would throw an exeption anyway.

Reply:
This change is rollbacked